.github/workflows/format.yml

```diff
@@ -7,17 +7,24 @@ on:
       - synchronize
       - ready_for_review
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
 jobs:
+  determine_runner:
+    if: github.event.pull_request.draft == false
+    name: Determine runner type to use
+    uses: ./.github/workflows/determine-workflow-runner.yml
+    with:
+      default_runner: ubuntu-latest

```

Comment:
I'm not sure it make sense to _potentially_ use the large runners for running `black` and `pylint`?

Reply:
Would you say we should exclude format/docs_check/upload to codecov from large runners?

Reply:
Excluding `upload-to-codecov` and `format` makes sense to me. I think including `docs` has some merit, considering that it's actually very useful for new features or big refactors to verify that the sphinx build passes without issues.

Reply:
The more I think about it, the more sense it makes to me to use the large runners for the actions above. If a PR is marked as urgent, we should treat all workflows as such.